sema4ai/tests/sema4ai_code_tests/test_vscode_integration.py

```diff
@@ -2364,3 +2364,40 @@ def test_create_agent_package(
     assert not result["success"]
     assert "Error creating Agent package" in result["message"]
     assert "not empty" in result["message"]
+
+    import yaml
+
+    # New YAML content to override the existing one
+    new_yaml_content = {
+        "agent-package": {
+            "spec-version": "v1",
+            "agents": [
+                {
+                    "name": "New Agent",
+                    "description": "Agent description",
+                    "model": "GPT 4 Turbo",
+                    "type": "agent",
+                    "reasoningLevel": 0,
+                    "runbooks": {"system": "system.md", "retrieval": "retrieval.md"},
+                    "action-packages": [],
+                    "resources": [],
+                }
+            ],
+        }
+    }
+
+    with open(f"{target_directory}/agent-spec.yaml", "w") as yaml_file:
+        yaml.dump(new_yaml_content, yaml_file, default_flow_style=False)
+
+    result = language_server.execute_command(
```

Comment:
I guess it may be nice to have one additional call with a bad agent-spec to verify that the message shown when validation fails is reasonable.